deepspeed/accelerator/real_accelerator.py

```diff
@@ -0,0 +1,71 @@
+from .abstract_accelerator import DeepSpeedAccelerator
+
+ds_accelerator = None
+
+
+def _validate_accelerator(accel_obj):
+    assert isinstance(accel_obj, DeepSpeedAccelerator), \
+        f'{accel_obj.__class__.__name__} accelerator is not subclass of DeepSpeedAccelerator'
+
+    assert accel_obj.is_available(), \
+        f'{accel_obj.__class__.__name__} accelerator fails is_available() test'
+
+
+def get_accelerator():
+    global ds_accelerator
+    if ds_accelerator is None:
+        from deepspeed.accelerator.cuda_accelerator import CUDA_Accelerator
+        ds_accelerator = CUDA_Accelerator()
+        _validate_accelerator(ds_accelerator)
+    return ds_accelerator
```

Comment:
Does it mean model have to call set_accelerator() explicitly to set accelerator other than CUDA_Accelerator?   Would we add XPU_Accelerator detection code in get_accelerator() once we have the detection code ready?

Reply:
Yes, `set_accelerator()` should be explicitly called to set an arbitrary accelerator. We currently apply only the two tests in `_valdate_accelerator()` on the accelerator object. 

CUDA_Accelerator logic exists in `get_acclerator()` as a special case for backwards-compatibility with existing client codes that are cuda-based.